### PR TITLE
Remove all vestiges of python2 from khan-dotfiles.

### PR DESCRIPTION
**Pull request by csilvers**
## Summary:
We no longer use python2 anywhere!  (Well, we still use it in one
place: `make topic-icons`, anyone who wants to run that will now need
to install python2 manually.)  Let's remove it from khan-dotfiles.
That simplifies the dotfiles a fair bit, and also simplifies the
users' dev experience since the virtualenv was causing problems.

I did this by running `git grep -e 'python[^3]' -e 'virtualenv'`
and deleting anything that looked no longer relevant.

Issue: https://khanacademy.atlassian.net/browse/INFRA-9771

## Test plan:
TODO -- will need to test on new installs of both mac and linux, and I
have no idea how to do that!

### CODE REVIEW

File: linux-functions.sh [hunk 1]
```diff
@@ -122,23 +106,21 @@ EOF
     # TODO(benkraft): Pull the version we want from webapp somehow.
     # curl for various scripts (including setup.sh)
     sudo apt-get install -y git \
-        python-dev \
+        python-dev python3-dev python3-pip pipenv \
         python-mode python-setuptools \
+        gcc \
```
Comment (dbraley) [inline linux-functions.sh:111]: interesting, where'd this come from? Or perhaps more interestingly, why didn't we have it before? I'd wager it was included as a dependency or something?

**Review by dbraley — APPROVED** [review]
Cool. Seems good by me! Outstanding we don't need python 2 any more!

File: linux-functions.sh (hunk 1 above)
Comment (csilvers) [reply linux-functions.sh:111]: We had it before.  I just did some totally unnecessary reordering...

File: linux-functions.sh [hunk 2]
```diff
@@ -122,23 +106,21 @@ EOF
...
         nodejs=12* \
+        cargo cargo-doc \
         redis-server \
         docker \
         curl \
         jq \
-        unzip tree sysdig iftop tcpflow \
-        cargo cargo-doc \
-        python-dev \
-        gcc \
```
Comment (csilvers) [inline linux-functions.sh:140]: (Previous location.)

File: linux-functions.sh (hunk 1 above)
Comment (dbraley) [reply linux-functions.sh:111]: Doh! I thought I did a match on all the packages and this was the only discrepancy not related to removing python 2. Guess I must have missed the original, sorry for any confusion!

**Comment by StevenACoffman** [comment]
Neat! It looks like after this, the only vestige requiring rosetta homebrew (brew86) is https://github.com/protocolbuffers/protobuf/releases/download/v3.4.0/protoc-3.4.0-osx-x86_64.zip which came out in [2017](https://github.com/protocolbuffers/protobuf/releases/v3.4.0).

The first release protoc release that has ARM64 (aarch_64) appears to be [Mar 25, 2022](https://github.com/protocolbuffers/protobuf/releases/tag/v3.20.0) which has https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch_64.zip

**Comment by csilvers** [comment]
> Neat! It looks like after this, the only vestige requiring rosetta homebrew (brew86) is https://github.com/protocolbuffers/protobuf/releases/download/v3.4.0/protoc-3.4.0-osx-x86_64.zip which came out in [2017](https://github.com/protocolbuffers/protobuf/releases/v3.4.0).
> 
> The first release protoc release that has ARM64 (aarch_64) appears to be [Mar 25, 2022](https://github.com/protocolbuffers/protobuf/releases/tag/v3.20.0) which has https://github.com/protocolbuffers/protobuf/releases/download/v3.20.0/protoc-3.20.0-osx-aarch_64.zip

Funny, I was just looking at our comment around that today for an unrelated reason:
```
 	// NOTE: this must stay pinned to the 20180205 version, at
	// least until we update all our .proto files so they no
	// longer use deprecated features that are removed in more
	// modern versions of this library.
	github.com/GoogleCloudPlatform/protoc-gen-bq-schema v0.0.0-20180205023744-481d6bbb8207
```
and was thinking, "it's a code generator, it's working fine.  There's no reason to upgrade it ever!"  Shows what I know.

**Pull request merged**